src/bflib_video.c

```diff
@@ -968,6 +968,18 @@ TbPixel LbPaletteFindColour(const unsigned char *pal, unsigned char r, unsigned
     }
     return *o;
 }
+
+/**
+ * Scale a given value according to the game resolution (units_per_pixel).
+ * default is based on 640x400 (units_per_pixel = 16)
+ * so if units_per_pixel = 16 the this function returns (base_value * 1)
+ *
+ * @param base_value The value (or size) for the game running at 640x400.
+ */
+long scale_value_for_resolution(long base_value)
```

Comment:
Is this a good place to add this function? I will make use of it throughout the KFX codebase as I continue to fix the scaling issues.

Reply:
It looks like it video-related so why not. Actually it is hard to break big and complex things to logical block sometimes.

Reply:
Good.

If a better place becomes obvious, this can always be moved in the future.